schedule.py

```diff
@@ -423,6 +424,22 @@ def print_semester_timetable(solver, time_assign, times, profs_classes, professo
     print()
 
 
+# find more schedules by setting one variable
+# print the optimal ones with the same objective_value
+def find_all_schedule(model, variables, objective_value, print_schedule, var1, var2, var3):
```

Comment:
1. What is `objective_value`? Seems weird for the caller to say what the optimum should be. My version of this function tries everything and only keeps the solutions with the lowest objective (since we're always minimizing), then returns them in a list. The printing is done from `main()`.

2. `var1`, `var2`, `var3` is bad and unnecessary. Since you have the `variables` dictionary, you can just loop through that to set each one to `True` in the loop.

Reply:
The objective value came from the one optimal schedule it already found. The method can use the value as a standard, and only output the schedules that are as good as the optimal one.
I originally wrote the method to return a list of schedules, but it will take a while before it prints anything due to runtime, especially for timetable scheduling. I can change it back if this is okay? or add a limit on how many schedules it need to find?
If it prints from main(), the var1, var2, var3 don't need to be there.

Reply:
I think waiting to print is fine - this is not something that requires immediate feedback. I'm not sure that anyone using this will have a good idea of what a good objective value should be, so let's just return the best ones.

Reply:
Bump - this has yet to be fixed.

Reply:
Updated, the method takes a long time for timetable scheduling - I ran it for an hour and it didn't finish.

Reply:
interesting. why is this slower than before?

Reply:
I didn't test how long it will run before, I assumed that it needs an immediate response. So after it ran for ten minutes I just switched to include the print method in it. 
Though I didn't expect it to be this slow since it is fast when applied to course assignment.

Reply:
Okay. We can merge this first, then see if there are ways to speed it up.